src/ArrayHelper.php

```diff
@@ -251,9 +249,9 @@ private static function getRootValue($array, $key, $default)
             try {
                 return $array::$$key;
             } catch (Throwable $e) {
-                // This is expected to fail if the property does not exist, or __get() is not implemented.
-                // It is not reliably possible to check whether a property is accessible beforehand.
-                return $array->$key;
+                return property_exists($array, (string) $key) || method_exists($array, '__get')
```

Comment:
1. That reduces performance because property_exists and property_exists use reflection internally.
2. For private properties property_exists return true but the property is not accessible.